### PR TITLE
JS: Add a js/client-side-url-redirect sink/source for `importScripts`. 

### PR DESCRIPTION
**Pull request by erik-krogh**
Adds support for the sink and source in CVE-2020-14000. 

[importScripts()](https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/importScripts) is a global function in WebWorkers that is used for importing scripts.  
(It's [used quite a lot](https://lgtm.com/query/4852283910000294627/)). 

Calling `importScripts(url)` does the same as the below `document.createElement(..)` snippet, but in a WebWorker. 

```JavaScript
var el = document.createElement("script");
el.src = url;
document.body.appendChild(el); // NOT OK
/// ---- Basically the same as ----
importScripts(url); 
```

This PR also adds support for registering `postMessage(..)` handlers using `global.onmessage`.   
E.g. 
```JavaScript
self.onmessage = function (e) {
    importScripts(e); // NOT OK
}
```

### CODE REVIEW

File: change-notes/1.26/analysis-javascript.md [hunk 1]
```diff
@@ -32,6 +32,7 @@
 | Unused loop iteration variable (`js/unused-loop-variable`) | Fewer results | This query no longer flags variables in a destructuring array assignment that are not the last variable in the destructed array. |
 | Unsafe shell command constructed from library input (`js/shell-command-constructed-from-input`) | More results | This query now recognizes more commands where colon, dash, and underscore are used. |
 | Unsafe jQuery plugin (`js/unsafe-jquery-plugin`) | More results | This query now detects more unsafe uses of nested option properties. |
+| Client-side URL redirect (`js/client-side-unvalidated-url-redirection`) | More results | This query now recognizes some unsafes uses of `importScripts()` inside WebWorkers. |
```
Comment (esbena) [inline change-notes/1.26/analysis-javascript.md:35]:
```suggestion
| Client-side URL redirect (`js/client-side-unvalidated-url-redirection`) | More results | This query now recognizes some unsafe uses of `importScripts()` inside WebWorkers. |
```

File: javascript/ql/src/semmle/javascript/security/dataflow/ClientSideUrlRedirectCustomizations.qll [hunk 2]
```diff
@@ -132,6 +132,15 @@ module ClientSideUrlRedirect {
     }
   }
 
+  /**
+   * An argument to `importScripts(..)` - which is used inside `WebWorker`s to import new scripts - viewed as a `ScriptUrlSink`.
+   */
+  class ImportScriptsSink extends ScriptUrlSink {
+    ImportScriptsSink() {
+      this = DataFlow::globalVarRef("importScripts").getACall().getAnArgument()
```
Comment (esbena) [inline javascript/ql/src/semmle/javascript/security/dataflow/ClientSideUrlRedirectCustomizations.qll:140]: > importScripts() and self.importScripts() are effectively equivalent — both represent importScripts() being called from inside the worker's inner scope.
> (https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/importScripts)

Do we support the mentioned `self.importScripts` as well?

**Review by esbena — COMMENTED** [review]
LGTM. One question and a nit.

File: javascript/ql/src/semmle/javascript/security/dataflow/ClientSideUrlRedirectCustomizations.qll (hunk 2 above)
Comment (erik-krogh) [reply javascript/ql/src/semmle/javascript/security/dataflow/ClientSideUrlRedirectCustomizations.qll:140]: We should - I'll add a test for it.

**Review by esbena — APPROVED** [review]
(tests are failing though)

**Comment by erik-krogh** [comment]
> (tests are failing though)

They just needed a restart.

**Pull request merged**